### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by ekexium**
Update dependency, limit and add simple code examples.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,33 +8,56 @@
...
+raft-proto = { git = "https://github.com/tikv/raft-rs", rev = "e624c1d48460940a40d8aa69b5329460d9af87dd" }
 ```
 
 The client requires a Git dependency until we can [publish it](https://github.com/tikv/client-rust/issues/32).
 
-There are [examples](examples) which show how to use the client in a Rust program.
+The client provides two modes to interact with TiKV: raw and transactional. 
+In the current version (0.0.0), transactional mode only supports optimistic transactions, which means commit may fail.
+
+Important note: It is **not recommended or supported** to use both the raw and transactional APIs on the same keyspace.
```
Comment (nrc) [inline README.md:32]:
```suggestion
Important note: It is **not recommended or supported** to use both the raw and transactional APIs on the same database.

```

File: README.md [hunk 2]
```diff
@@ -8,33 +8,56 @@
...
+
+[patch.crates-io]
+raft-proto = { git = "https://github.com/tikv/raft-rs", rev = "e624c1d48460940a40d8aa69b5329460d9af87dd" }
 ```
 
 The client requires a Git dependency until we can [publish it](https://github.com/tikv/client-rust/issues/32).
 
-There are [examples](examples) which show how to use the client in a Rust program.
+The client provides two modes to interact with TiKV: raw and transactional. 
+In the current version (0.0.0), transactional mode only supports optimistic transactions, which means commit may fail.
```
Comment (nrc) [inline README.md:30]:
```suggestion
In the current version (0.0.0), the transactional API only supports optimistic transactions.

```

File: src/raw/client.rs [hunk 3]
```diff
@@ -8,6 +8,7 @@ use tikv_client_common::{BoundRange, Config, Error, Key, KvPair, Result, Value};
 const MAX_RAW_KV_SCAN_LIMIT: u32 = 10240;
 
 /// The TiKV raw [`Client`](Client) is used to issue requests to the TiKV server and PD cluster.
+/// The client should not be used with the transactional client on the same keyspace, since they use different encodings.
```
Comment (nrc) [inline src/raw/client.rs:11]: I would not add these two warnings, the one in the README is enough, IMO

**Review by nrc — APPROVED** [review]

**Comment by ekexium** [comment]
@nrc I added some content. Could you take another look?

File: README.md [hunk 4]
```diff
@@ -8,41 +8,123 @@
...
+| Request     | Main parameter type | Successful result type | Noteworthy Behavior                           |
+| ----------- | ------------------- | ---------------------- | --------------------------------------------- |
+| `put`       | `KvPair`            | `()`                   |                                               |
+| `get`       | `Key`               | `Option<value>`        |                                               |
+| `delete`    | `Key`               | `()`                   |                                               |
+| `scan`      | `BoundRange`        | `Iter<KvPair>`         |                                               |
+| `batch_get` | `Iter<Key>`         | `Iter<KvPair>`         | Skip non-existent keys; Does not retain order |
+| `lock_keys` | `KvPair`            | `()`                   |                                               |
+
+For detailed behavior of each reqeust, please refer to the [doc](#Access-the-documentation).
```
Comment (nrc) [inline README.md:84]:
```suggestion
For detailed behavior of each request, please refer to the [doc](#Access-the-documentation).

```

**Review by nrc — APPROVED** [review]
lgtm, just one typo, then land it.

re batch_scan/each_limit, is this behaviour specific to the Rust client? Does something else happen in TiDB?

File: README.md (hunk 4 above)
Comment (ekexium) [reply README.md]: `batch_scan` is a raw request. So TiDB should not have this problem.

**Pull request merged**